libs/elasticsearch/langchain_elasticsearch/cache.py

```diff
@@ -0,0 +1,193 @@
+import hashlib
+import logging
+from datetime import datetime
+from functools import cached_property
+from typing import TYPE_CHECKING, Any, Dict, Optional
+
+import elasticsearch
+from elasticsearch import NotFoundError
+from langchain_core.caches import RETURN_VAL_TYPE, BaseCache
+from langchain_core.load import dumps, loads
+
+from langchain_elasticsearch.client import create_elasticsearch_client
+
+if TYPE_CHECKING:
+    from elasticsearch import Elasticsearch
+
+logger = logging.getLogger(__name__)
+
+
+class ElasticsearchCache(BaseCache):
+    """An Elasticsearch cache integration for LLMs."""
+
+    def __init__(
+        self,
+        es_index: str,
```

Comment:
Could we name this `index_name` to be consistent with the other classes?

Reply:
done